lib/registry/src/lib.rs

```diff
@@ -894,6 +894,48 @@ pub fn install_package(
     ))
 }
 
+pub fn whoami(
+    #[cfg(test)] test_name: &str,
+    registry: Option<&str>,
+) -> Result<(String, String), anyhow::Error> {
+    use crate::graphql::{who_am_i_query, WhoAmIQuery};
+    use graphql_client::GraphQLQuery;
+
+    #[cfg(test)]
+    let config = PartialWapmConfig::from_file(test_name);
+    #[cfg(not(test))]
+    let config = PartialWapmConfig::from_file();
+
+    let config = config
+        .map_err(|e| anyhow::anyhow!("{e}"))
+        .context(anyhow::anyhow!("{registry:?}"))?;
```

Comment:
We can drop the `map_err()` because the `context()` method should automatically wrap an error in an `anyhow::Error`. 

Also, we need to use `with_context()` instead of `context()` if constructing the context requires function calls. This is similar to [Clippy's lint](https://rust-lang.github.io/rust-clippy/master/index.html#or_fun_call) about unconditionally making expensive calls with side-effects when they should only be triggered during an error.

```suggestion
        .with_context(|| format!("{registry:?}"))?;
```